### PR TITLE
VideoPress: Clean up REST API endpoint schemas

### PR DESCRIPTION
**Pull request by obenland**
## Proposed changes:
* Use proper JSON Schema types (`integer` instead of `int`) and built-in constraints (`enum`) instead of custom `validate_callback` functions
* Remove redundant `'required' => false` (it's the default) and `'sanitize_callback' => 'rest_sanitize_boolean'` (handled by `'type' => 'boolean'`)
* Add explicit arg definitions for URL path parameters (`video_guid`, `post_id`)
* Tighten route regexes from `\w+` to `[A-Za-z0-9]{8}` so GUID format is enforced at the routing level, making schema-level `pattern` unnecessary
* Add `esc_attr()` to regex-extracted block anchor for consistency with the existing block attributes path
* Tighten AJAX GUID validation to exactly 8 alphanumeric characters, matching `videopress_is_valid_guid()`
* Clean up tests: validate against actual registered schemas, remove unused code, add route-level GUID dispatch tests

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
* Verify VideoPress video embeds still render correctly on the frontend (both iframe and oEmbed paths)
* Verify the VideoPress block in the editor still allows setting privacy (0=Public, 1=Private, 2=Site Default) and rejects invalid values
* Verify video poster updates work (GET and POST to the poster endpoint)
* Run `pnpm jetpack test php packages/videopress` — all tests should pass

### CODE REVIEW

File: projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php [hunk 1]
```diff
@@ -113,6 +103,14 @@ public function register_routes() {
 			$this->namespace,
 			$this->rest_base . '/(?P<video_guid>\w+)/poster',
 			array(
+				'args' => array(
+					'video_guid' => array(
+						'description' => __( 'The VideoPress GUID.', 'jetpack-videopress-pkg' ), // @phan-suppress-current-line PhanPluginMixedKeyNoKey
+						'type'        => 'string',
+						'required'    => true,
+						'pattern'     => '[a-zA-Z0-9]{8}',
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php]: ~~Length is already constrained by the route regex.~~

Copilot was right — `\w+` doesn't constrain length. Fixed in 1700caf079: tightened route regex to `[A-Za-z0-9]{8}` and removed the schema `pattern` since the route regex now handles validation.

File: projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php [hunk 2]
```diff
@@ -159,6 +147,19 @@ public function register_routes() {
 			$this->namespace,
 			$this->rest_base . '/(?P<video_guid>\w+)/check-ownership/(?P<post_id>\d+)/',
 			array(
+				'args' => array(
+					'video_guid' => array(
+						'description' => __( 'The VideoPress GUID.', 'jetpack-videopress-pkg' ), // @phan-suppress-current-line PhanPluginMixedKeyNoKey
+						'type'        => 'string',
+						'required'    => true,
+						'pattern'     => '[a-zA-Z0-9]{8}',
+					),
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php:154]: ~~Same as above — the route regex constrains length.~~

Fixed in 1700caf079 — see above.

File: projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php [hunk 3]
```diff
@@ -182,11 +183,19 @@ public function register_routes() {
...
 		register_rest_route(
 			$this->namespace,
 			$this->rest_base . '/playback-jwt/(?P<video_guid>\w+)',
 			array(
+				'args'                => array(
+					'video_guid' => array(
+						'description' => __( 'The VideoPress GUID.', 'jetpack-videopress-pkg' ),
+						'type'        => 'string',
+						'required'    => true,
+						'pattern'     => '[a-zA-Z0-9]{8}',
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php]: ~~Same as above — the route regex constrains length.~~

Fixed in 1700caf079 — see above.

File: projects/packages/videopress/src/class-initializer.php [hunk 4]
```diff
@@ -326,7 +326,7 @@ public static function render_videopress_video_block( $block_attributes, $conten
 			$maybe_premium_script = '<script>' . $script_content . '</script>';
 		}
 
-		// $id_attribute, $video_wrapper, $figcaption properly escaped earlier on the code
+		// $id_attribute, $video_wrapper, $figcaption properly escaped earlier on the code.
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/src/class-initializer.php]: Fixed in 364cb3df1c.

File: projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+		}
+		return null;
+	}
+
+	/**
+	 * Test that all expected REST routes are registered.
+	 */
+	public function test_routes_are_registered() {
+		$this->assertNotNull( $this->find_route_key( 'videopress/meta' ) );
+		$this->assertNotNull( $this->find_route_key( 'videopress/' ) && $this->find_route_key( '/poster' ) );
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php]: Fixed in 364cb3df1c — split into separate assertions.

File: projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+		$routes     = rest_get_server()->get_routes();
+		$route_data = $routes['/wpcom/v2/videopress/playback-jwt/(?P<video_guid>\\w+)'];
+		$args       = $route_data[0]['args'];
+
+		$pattern = $args['video_guid']['pattern'];
+
+		if ( $expected ) {
+			$this->assertMatchesRegularExpression( '/^' . $pattern . '$/', $guid );
+		} else {
+			$this->assertDoesNotMatchRegularExpression( '/^' . $pattern . '$/', $guid );
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php]: Fixed in 364cb3df1c — now uses `rest_validate_value_from_schema()`. Also replaced the "too long" test case with "empty string" since length is enforced by the route regex, not the schema pattern.

File: projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php]: Fixed in 1700caf079 — tightened route regex to `[A-Za-z0-9]{8}` and removed the schema `pattern` entirely since the route regex handles validation.

File: projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php:154]: Same as above — fixed in 1700caf079.

File: projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php]: Same as above — fixed in 1700caf079.

File: projects/packages/videopress/src/class-initializer.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/src/class-initializer.php]: Fixed in 8300ab1102.

File: projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php]: Fixed in 1700caf079 — replaced with `assertArrayHasKey` on exact route keys.

File: projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php]: Fixed in 1700caf079 — tests now validate against the actual registered route schema. The pattern-based test was replaced with route-level dispatch tests.

File: projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php]: Fixed in 1700caf079 — route regex tightened to `[A-Za-z0-9]{8}` and schema `pattern` removed.

File: projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+	/**
+	 * Set up the test environment.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		global $wp_rest_server;
+		$wp_rest_server = new WP_REST_Server();
+		$this->server   = $wp_rest_server;
+
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php:36]: Fixed in 1700caf079 — removed the property and assignment.

File: projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+	 *
+	 * @return array[] Test cases.
+	 */
+	public static function video_guid_provider(): array {
+		return array(
+			'valid 8-char alphanumeric' => array( 'AbCd1234', true ),
+			'valid all lowercase'       => array( 'abcd1234', true ),
+			'valid all uppercase'       => array( 'ABCD1234', true ),
+			'valid all digits'          => array( '12345678', true ),
+			'too short'                 => array( 'abc1234', false ),
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php]: Fixed in 1700caf079 — replaced schema-level pattern tests with route-level dispatch tests (`test_route_rejects_invalid_video_guid` and `test_route_accepts_valid_video_guid`) that verify the route regex itself.

File: projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+ * Validates that REST route schemas correctly enforce argument constraints.
+ */
+class WPCOM_REST_API_V2_Endpoint_VideoPress_Test extends BaseTestCase {
+
+	/**
+	 * Full route keys used across tests.
+	 */
+	private const ROUTE_META            = '/wpcom/v2/videopress/meta';
+	private const ROUTE_POSTER          = '/wpcom/v2/videopress/(?P<video_guid>[A-Za-z0-9]{8})/poster';
+	private const ROUTE_CHECK_OWNERSHIP = '/wpcom/v2/videopress/(?P<video_guid>[A-Za-z0-9]{8})/check-ownership/(?P<post_id>\d+)';
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/tests/php/WPCOM_REST_API_V2_Endpoint_VideoPress_Test.php:24]: WordPress strips the trailing slash when registering routes, so the constant without it matches the actual registered key. Tests pass as-is.

File: projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php [hunk 10]
```diff
@@ -111,8 +101,15 @@ public function register_routes() {
...
-			$this->rest_base . '/(?P<video_guid>\w+)/poster',
+			$this->rest_base . '/(?P<video_guid>[A-Za-z0-9]{8})/poster',
 			array(
+				'args' => array(
+					'video_guid' => array(
+						'description' => __( 'The VideoPress GUID.', 'jetpack-videopress-pkg' ), // @phan-suppress-current-line PhanPluginMixedKeyNoKey
+						'type'        => 'string',
+						'required'    => true,
+					),
+				),
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/packages/videopress/src/class-wpcom-rest-api-v2-endpoint-videopress.php:112]: PR description updated to reflect the current approach — validation is handled by the route regex, not schema `pattern`.
